libopenage/gui/guisys/private/gui_event_queue_impl.cpp

```diff
@@ -24,7 +28,12 @@ GuiEventQueueImpl* GuiEventQueueImpl::impl(GuiEventQueue *event_queue) {
 
 void GuiEventQueueImpl::process_callbacks() {
 	assert(QThread::currentThread() == this->thread);
+#ifdef __APPLE__
+	if (QThread::currentThread() != QCoreApplication::instance()->thread())
+		this->callback_processor.processEvents();
+#else
 	this->callback_processor.processEvents();
```

Comment:
why does apple need a special treatment here? (maybe even add a comment about the reason)